components/brave_wallet/renderer/brave_wallet_js_handler.cc

```diff
@@ -616,9 +568,18 @@ void BraveWalletJSHandler::InjectInitScript() {
 
 void BraveWalletJSHandler::FireEvent(const std::string& event,
                                      base::Value event_args) {
-  base::Value args = base::Value(base::Value::Type::LIST);
-  args.Append(event);
-  args.Append(std::move(event_args));
```

Comment:
Did we use it wrong here? As it used to work fine via old way

Reply:
no, it is just moved to a shared function in this commit
https://github.com/brave/brave-core/commit/a0bb93b0379db76a8be809d180f71e210f01c293

Reply:
and it is now takes `std::vector<v8::Local<v8::Value>>` instead of `std::vector<base::Value>` with doing v8 conversion inside because we have use case like emitting a v8 object instead of creating v8 object from value